### PR TITLE
tests: use in-tree snap{ctl,-exec} for all tests

### PR DESCRIPTION
**Pull request by mvo5**

### CODE REVIEW

**Comment by chipaca** [comment]
:+1:

**Comment by kyrofa** [comment]
You can probably remove the prepare from the snap-run-hook test as well.

**Comment by kyrofa** [comment]
But other than that, :+1: .

**Comment by mvo5** [comment]
Thanks, I removed the prepare  for snap-run-hook now.

File: tests/lib/prepare.sh [hunk 1]
```diff
@@ -2,6 +2,41 @@
...
 set -eux
 
+update_core_snap_with_snap_exec_snapctl() {
+    # We want to use the in-tree snap-exec and snapctl, not the ones in the core
+    # snap. To accomplish that, we'll just unpack the core we just grabbed,
+    # shove the new snap-exec and snapctl in there, and repack it.
+
+    # First of all, unmount the core
+    core="$(realpath /snap/ubuntu-core/current)"
+    snap="$(mount | grep "$core" | awk '{print $1}')"
```
Comment (kyrofa) [inline tests/lib/prepare.sh]: If you change this line to `snap="$(mount | grep " $core" | awk '{print $1}')"` (notice the space before `$core`) it'll work on ubuntu-core.

**Comment by kyrofa** [comment]
There's no restore procedure in this PR. Do we need one? The two spread tests you've updated have one... should they be removed?

**Comment by niemeyer** [comment]
Tests broken, conflicts, some questions open.

**Comment by mvo5** [comment]
@kyrofa Thanks for your review. About the restore - it seems we only needed that to restore the pristine ubuntu-core for the other tests. However it seems like that by applying that change (in-tree snap-exec/snapctl) we don't need the restore anymore. Or am I misunderstanding the question?

**Comment by mvo5** [comment]
Test fails because of ` Fetch and check assertions for snap "ubuntu-core" (423) (Get https://assertions.ubuntu.com/v1/assertions/snap-revision/7sS9MbzjeZ87E4IVq3ByrAO8t1YqriNlaX4r7eT5AHYZjzRCa7twTmWU53e6falH: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers))` which looks unrelated.

**Comment by kyrofa** [comment]
> @kyrofa Thanks for your review. About the restore - it seems we only needed that to restore the pristine ubuntu-core for the other tests. However it seems like that by applying that change (in-tree snap-exec/snapctl) we don't need the restore anymore. Or am I misunderstanding the question?

Indeed, that makes sense. Though that probably means the restore logic in the specific tests should probably be removed then, or they'll try to get back to the pristine core again, no?

**Comment by kyrofa** [comment]
> Indeed, that makes sense. Though that probably means the restore logic in the specific tests should probably be removed then, or they'll try to get back to the pristine core again, no?

Ah, you did since my previous review, sorry! That looks better. You probably also want to update the `snap-set` test. I also see that this is only happening for classic, but now that you fixed [this](https://github.com/snapcore/snapd/pull/1872/files/953e7732c6545419636f95c77d2e3fb6d9f5f8df#r78040763) it works with ubuntu-core now-- worth making it happen for both?

**Comment by mvo5** [comment]
@kyrofa Thanks! I updated the snap-set test now too. We also use the in-tree snap-exec/snapctl for the image tests, see https://github.com/snapcore/snapd/blob/master/tests/lib/prepare.sh#L79 - unless I missunderstood your comment of course :)

**Pull request merged**